node/flatpak_node_generator/providers/__init__.py

```diff
@@ -45,32 +46,48 @@ def process_lockfile(self, lockfile: Path) -> Iterator[Package]:
         raise NotImplementedError()
 
 
-class RCFileProvider:
-    RCFILE_NAME: str
+@dataclass
+class Config:
+    data: Dict[str, Any] = dataclasses.field(default_factory=lambda: {})
 
-    def parse_rcfile(self, rcfile: Path) -> Dict[str, str]:
-        with open(rcfile, 'r') as r:
-            rcfile_text = r.read()
-        parser_re = re.compile(
-            r'^(?!#|;)(\S+)(?:\s+|\s*=\s*)(?:"(.+)"|(\S+))$', re.MULTILINE
-        )
-        result: Dict[str, str] = {}
-        for key, quoted_val, val in parser_re.findall(rcfile_text):
-            result[key] = quoted_val or val
-        return result
-
-    def get_node_headers(self, rcfile: Path) -> Optional[NodeHeaders]:
-        rc_data = self.parse_rcfile(rcfile)
-        if 'target' not in rc_data:
+    def merge_new_keys_only(self, other: Dict[str, Any]) -> None:
+        for key, value in other.items():
+            if key not in self.data:
+                self.data[key] = value
+
+    def get_node_headers(self) -> Optional[NodeHeaders]:
+        if 'target' not in self.data:
             return None
-        target = rc_data['target']
-        runtime = rc_data.get('runtime')
-        disturl = rc_data.get('disturl')
+        target = self.data['target']
+        runtime = self.data.get('runtime')
+        disturl = self.data.get('disturl')
 
         assert isinstance(runtime, str) and isinstance(disturl, str)
 
         return NodeHeaders.with_defaults(target, runtime, disturl)
 
+    def get_registry_for_scope(self, scope: str) -> Optional[str]:
+        return self.data.get(f'{scope}:registry')
+
+
+class ConfigProvider:
+    @property
+    def _filename(self) -> str:
```

Comment:
Is there any reason for the filename to be a property and not a class variable? I can't imagine npm/yarn config file name ever changing.

Reply:
It's mostly because then we could use actual abstractproperty handling in the future, but atm it does mean that forgetting to override it gives a clearer error at the call site.